### ISSUE TITLE
Shouldn't the state be kept?

### ISSUE THREAD

**Issue by ziame** [robots-from-jupyter/robotkernel#21]
When using Jupyter Notebook, its main pull for me was the possibility of keeping the state between cells. But with Robot Kernel, when I have a cell:

`*** TASKS ***
Setting Variable
    ${variable}    Set Variable    stuff`

I cannot access it in the next cell:
`*** TASKS ***
Setting Variable
    Log ${variable}`

This of course is linked to test scope and everything, but settings Suite Variables also fails, because every run "restarts" the suite. Is there some other way of going about this? Maybe i'm doing something wrong, but I wasn't able to get it to work without readding *** TASKS *** to every next cell.

It also does not work with global scope libraries. I import a global library, set a state to in one task, but the next task 'restarts' everything and the state is not kept.

tl;dr:
i would expect to be able to set state in one cell and have it kept in the next cell.

**Comment by datakurre** [comment]
No and yes. Currently it does not keep variable state. And that’s because of the execution model. I heard that Robot Framework has API for programmatic execution to make it “keep state”, but I cannot even estimate yet how much work that would be.

Yet, I could “simulate” this feature by saving suite variables after cell execution and injecting them before the next cell execution. I’d expect to be able to do this within a few next weeks, but I need to finish the widget support currently in master. I have been bitten by this also myself, so thank you for pointing this out.

**Comment by datakurre** [comment]
Keepin state of global variables in now on master, but I'm still stuck with some widget related issues before releasing

https://github.com/robots-from-jupyter/robotkernel/commit/fb506cabe6ef61e8bb8ae72650948777710bc78b

**Comment by datakurre** [comment]
This should be fixed in 0.12.1
